src/retryable.rs

```diff
@@ -0,0 +1,128 @@
+use std::{future::Future, time::Duration};
+
+#[cfg(feature = "async")]
+use async_std::task::sleep;
+#[cfg(feature = "async")]
+use reqwest::{RequestBuilder, Response as ReqwestResponse, StatusCode};
+
+#[cfg(feature = "blocking")]
+use ureq::{Error as UReqError, Request, Response as UReqResponse};
+
+const SECOND: u64 = 1_000;
+const BASE_BACKOFF_MS: u64 = 5_000;
+const DEFAULT_MAX_RETRIES: u32 = 5;
+
+pub const RETRY_TOO_MANY_REQUEST_ONLY: [StatusCode; 1] = [StatusCode::TOO_MANY_REQUESTS];
+
+pub trait AsyncRetryable<R, E = crate::Error> {
+    fn exec_with_retry(
+        self,
+        retryable_error_codes: Vec<StatusCode>,
+        max_retries: Option<u32>,
+    ) -> impl Future<Output = Result<R, E>>;
+}
+
+pub trait SyncRetryable<R, E = crate::Error> {
+    fn exec_with_retry(
+        self,
+        retryable_error_codes: Vec<StatusCode>,
+        max_retries: Option<u32>,
+    ) -> Result<R, E>;
+}
+
+fn compute_backoff(retry_after: Option<&str>, backoff: Duration) -> Duration {
+    match retry_after {
+        // If response has a Retry-After header, parse it and use it as backoff
+        Some(retry_after) => {
+            return Duration::from_millis(
+                retry_after
+                    .parse::<u64>()
+                    .map(|seconds| seconds * SECOND)
+                    .unwrap_or(BASE_BACKOFF_MS),
+            )
+        }
+        // Else, double previous backoff
+        _ => return backoff * 2,
```

Comment:
Is there a reason why this is double the previous backoff? I've got nothing against it, just wondering why this specific number.

Reply:
Exponentially increasing the backoff is standard, to reduce 429 probability as errors come in (the more you get 429s, the more you are expected to be jailed by server, so you need to wait more). 

On the other hand, the factor 2 is quite arbitrary, I'm open to modify this value

Reply:
worth noting that we always prefer using explicit Retry-After header over this to set backoff